beta-src/src/components/ui/WDInfoPanel.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+import WDPopover from "./WDPopover";
+import WDCountryTable from "./WDCountryTable";
+import WDVoteButtons from "./WDVoteButtons";
+import { CountryTableData } from "../../interfaces/CountryTableData";
+import { UserData } from "../../interfaces/UserData";
+import getDevice from "../../utils/getDevice";
+
+interface WDInfoPanelProps {
+  countries: CountryTableData[];
+  userData: UserData;
+}
+
+const WDInfoPanel: React.FC<WDInfoPanelProps> = function ({
+  countries,
+  userData,
+}): React.ReactElement {
+  const [voteState, setVoteState] = React.useState(userData.votes);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const device = getDevice({ width: 1300, height: 900 });
+
+  const toggleVote = (voteName: string, voteStatus: boolean) => {
+    const newVoteState = {
+      ...voteState,
+      [voteName]: !voteStatus,
+    };
+
+    setVoteState(newVoteState);
+  };
```

Comment:
Why do you need to pass in voteStatus here? Don't you have access to the vote state?

Reply:
I believe we should avoid referencing current state within setState.

Reply:
I don't think we need to avoid doing something like this, we are still referencing the state within the custom toggleVote function. It also aligns better with the title of the function `toggleVote` which is different from `setVote`, toggle indicates only two possible values. Eg. on/off 
```
const toggleVote = (voteName: string) => {
    const newVoteState = {
      ...voteState,
      [voteName]: !voteState[voteName],
    };

    setVoteState(newVoteState);
  };
```

Reply:
it was Vote enum when i first build out the VoteButton components. However, when I applied the map. to render the three buttons, I had to change to string due to the reason that i need to loop through the array to get the vote

const commandButtons = Object.entries(voteState).map(([vote, status]) => {
    return (
      <WDButton
        key={vote}
        color={status ? "secondary" : "primary"}
        onClick={() => toggleVote(vote)}
        startIcon={status ? <WDCheckmarkIcon /> : ""}
      >
        {voteLabel[vote]}
      </WDButton>
    );
  });